### PR TITLE
Add method to iterate over brains in upgrade step.

### PR DESCRIPTION
**Pull request by njohner**
I'm not sure why we don't have a method to iterate over the brains. We don't always want to get the objects.

For https://4teamwork.atlassian.net/browse/CA-2785

### CODE REVIEW

File: ftw/upgrade/step.py [hunk 1]
```diff
@@ -98,6 +98,21 @@ def objects(self, catalog_query, message, logger=None,
...
+    def brains(self, catalog_query, message, logger=None,
+               savepoints=None):
+        """Queries the catalog (unrestricted) and creates an iterator
+        over the brains.
+        The iterator configures and calls a ``ProgressLogger`` with the
+        passed ``message``.
+        """
+
+        brains = self.catalog_unrestricted_search(
+            catalog_query, full_objects=False)
```
Comment (deiferni) [inline ftw/upgrade/step.py:111]: you could add some code re-use for the existing `objects` method and this new `brains` method. The only thing that is different is the `full_objects` argument.

File: ftw/upgrade/step.py [hunk 2]
```diff
@@ -98,6 +98,21 @@ def objects(self, catalog_query, message, logger=None,
         objects = SavepointIterator.build(objects, savepoints, logger)
         return ProgressLogger(message, objects, logger=logger)
 
+    security.declarePrivate('brains')
+    def brains(self, catalog_query, message, logger=None,
```
Comment (deiferni) [inline ftw/upgrade/step.py:102]: can you provide some tests for this method?

**Review by deiferni — APPROVED** [review]
LGTM 👍 

please do add tests though.

**Comment by njohner** [comment]
All done...
I've also realised that some tests were being run twice, so I corrected that as well.

**Pull request merged**